src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const separator = toFormat[toFormat.length - 1];
+
+  const dateArr = date.split(oldSeparator);
+
+  let yearIndex;
+
+  if (fromFormat.indexOf('YYYY') !== -1) {
+    yearIndex = fromFormat.indexOf('YYYY');
+  } else {
+    yearIndex = fromFormat.indexOf('YY');
+  }
+
+  const monthIndex = fromFormat.indexOf('MM');
+  const dayIndex = fromFormat.indexOf('DD');
+
+  let year = dateArr[yearIndex];
+  const month = dateArr[monthIndex];
+  const day = dateArr[dayIndex];
+
+  if (year.length === 2 && toFormat.includes('YYYY')) {
+    if (year < 30) {
+      year = `20${year}`;
+    } else {
+      year = `19${year}`;
+    }
```

Comment:
The logic for converting two-digit years assumes that years less than 30 belong to the 2000s and others to the 1900s. This could lead to incorrect results if the input data doesn't follow this assumption. Consider making this logic configurable or more flexible.